src/ImageSharp/Formats/Tiff/TiffFrameMetadata.cs

```diff
@@ -0,0 +1,370 @@
+// Copyright (c) Six Labors.
+// Licensed under the Apache License, Version 2.0.
+
+using System.Collections.Generic;
+using System.Linq;
+using SixLabors.ImageSharp.Formats.Experimental.Tiff.Constants;
+using SixLabors.ImageSharp.Metadata;
+using SixLabors.ImageSharp.Metadata.Profiles.Exif;
+
+namespace SixLabors.ImageSharp.Formats.Experimental.Tiff
+{
+    /// <summary>
+    /// Provides Tiff specific metadata information for the frame.
+    /// </summary>
+    public class TiffFrameMetadata : IDeepCloneable
+    {
+        // 2 (Inch)
+        internal const ushort DefaultResolutionUnit = 2;
+
+        private const TiffPlanarConfiguration DefaultPlanarConfiguration = TiffPlanarConfiguration.Chunky;
+
+        private const TiffPredictor DefaultPredictor = TiffPredictor.None;
+
+        private ExifProfile frameTags;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="TiffFrameMetadata"/> class.
+        /// </summary>
+        public TiffFrameMetadata()
+        {
+        }
+
+        /// <summary>
+        /// Gets the Tiff directory tags.
+        /// </summary>
+        public ExifProfile ExifProfile
+        {
+            get => this.frameTags ??= new ExifProfile();
+            internal set => this.frameTags = value;
+        }
+
+        /// <summary>
+        /// Gets a general indication of the kind of data contained in this subfile.
+        /// </summary>
+        /// <value>A general indication of the kind of data contained in this subfile.</value>
+        public TiffNewSubfileType SubfileType => (TiffNewSubfileType?)this.ExifProfile.GetValue(ExifTag.SubfileType)?.Value ?? TiffNewSubfileType.FullImage;
+
+        /// <summary>
+        /// Gets a general indication of the kind of data contained in this subfile.
+        /// </summary>
+        /// <value>A general indication of the kind of data contained in this subfile.</value>
+        public TiffSubfileType? OldSubfileType => (TiffSubfileType?)this.ExifProfile.GetValue(ExifTag.OldSubfileType)?.Value;
+
+        /// <summary>
+        /// Gets the number of columns in the image, i.e., the number of pixels per row.
+        /// </summary>
+        public Number Width
```

Comment:
I don't think these properties should throw. Metadata properties should be treated as supplementary data not required.

Reply:
I would look at the way metadata is used in the other formats. Each property should be settable with sensible defaults. 

We should pull the values in their entirety on decode allow adjustment and set on encode. I would also carefully consider the usefulness of Width/Height properties. They can immediately go out of sync with the image frame dimensions.

Reply:
> I don't think these properties should throw. Metadata properties should be treated as supplementary data not required.

I agree with that, but some are required to decode the image for example `BitsPerSample`. I think at least this one should throw because we cannot guess it good enough. The only exception is for this is when `TiffPhotometricInterpretation.WhiteIsZero` and `TiffPhotometricInterpretation.BlackIsZero`. Then its clear that it must be 1 bit per sample. For the other TiffPhotometricInterpretations, its ambiguous.

Im also not sure what todo with images where the image width and height is missing.

Reply:
Is BitsPerSample purely derived from the EXIF metadata? That’s an odd requirement if so

Reply:
> Is BitsPerSample purely derived from the EXIF metadata? That’s an odd requirement if so

I would divide the `Tiff` `tags` into 2 groups - *image properties* (`BitsPerSample`, `Compression`, `StripByteCounts` ...) and *pure metadata*. For example in `Jpeg` format, `ExifProfile` used only as metadata container, but in `Tiff` it also stores image properties. Which group the `Width` belongs to, I can't say...

Reply:
Ah ok, so I think I understand where the confusion comes from. Please correct me if I'm wrong.

TIFF uses IFDs to store required data about the image. They can be considered extensions to the Tiff Header. 

All IFDs are store in the EXIF format but only one of these IFDs (_ExifIFDPointer_) contains the pure metadata EXIF profile. 

So, no, the `Width` and `Height` should definitely not be stored in this class. They are properties that should be decoded to each `ImageFrame<TPixel>`. 

I would say that properties such as `BitsPerPixel`, `BitsPerSample`, `PhotometricInterpretation ` should be present since they can be used to determine how to encode an edited Tiff however, only the absolutely required properties should be stored and they should be standard properties with public getters and setters like the other formats. See [PngMetadata](https://github.com/SixLabors/ImageSharp/blob/36018bdfe6da67b79ca3be1e85a0dde718929467/src/ImageSharp/Formats/Png/PngMetadata.cs) for example. 

Sanitation of those properties should be the responsibility of the encoder.

`Compression` is an odd one? Can individual frames be compressed differently?

I've also noted property duplication in both `TiffMetadata` and this class. I'll need convincing that they are both required.

Reply:
@JimBobSquarePants yeah you got it right, yes, frames can be compressed differently

Reply:
@brianpopow see the comments here. This class needs to be rewritten to remove the several properties and reorganize the code to move any sanitation responsibility to the encoder.